### ISSUE TITLE
poetry add does a full dependency resolution scan to add a package with no dependencies

### ISSUE THREAD

**Issue by jgentil** [python-poetry/poetry#4034]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10 Version 20H2 Build 19042.928
- **Poetry version**: 1.1.6
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**:  https://gist.github.com/jgentil/c12b5e1cef6be9b34c103b0038163858
## Issue

Within an existing project that already has a working Poetry environment running, when I attempt to add a new package that has no dependencies of it's own, Poetry still goes through the full motions of resolving all other dependencies, which haven't changed at all since the last update.  I expected that just the dependencies that the package itself needed would be resolved and updated as necessary, which are none.  

The `--source pypi` is added because otherwise Poetry will attempt to search for each package on the secondary source in my project, which has none of these packages but is really slow to respond.  This in itself is annoying, as I would prefer Poetry to only use the secondary repository when it's explicitly told to.  

```
C:\Users\me\repos\poetrytest>poetry add semantic_version --source pypi -vvv
Using virtualenv: C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\poetrytest-wfEcoP71-py3.7
PyPI: 24 packages found for semantic-version *
Using version ^2.8.5 for semantic-version

Updating dependencies
Resolving dependencies...
   1: fact: poetrytest is 0.1
   1: derived: poetrytest
   1: fact: poetrytest depends on azure-storage-blob (12.6.0)
   1: fact: poetrytest depends on azure-identity (1.5.0b2)
   1: fact: poetrytest depends on paramiko (~2.7.2)
   1: fact: poetrytest depends on pyasn1 (~0.4.8)
   1: fact: poetrytest depends on pypac (~0.14.0)
   1: fact: poetrytest depends on keyring (~21.8.0)
   1: fact: poetrytest depends on PyJWT (~1.7.1)
   1: fact: poetrytest depends on pywebview (3.4)
   1: fact: poetrytest depends on importlib-metadata (~2.1.1)
   1: fact: poetrytest depends on pydantic (^1.8.1)
   1: fact: poetrytest depends on semantic-version (^2.8.5)
   1: fact: poetrytest depends on ipython (7.19.0)
   1: fact: poetrytest depends on pytest (6.1.2)
   1: fact: poetrytest depends on pytest-mock (3.3.1)
   1: fact: poetrytest depends on ipython (7.19.0)
   1: fact: poetrytest depends on pytest (6.1.2)
   1: fact: poetrytest depends on pytest-mock (3.3.1)
   1: fact: poetrytest depends on locallibrary (0.2.0)
   1: selecting poetrytest (0.1)
   1: derived: locallibrary (==0.2.0)
   1: derived: pytest-mock (==3.3.1)
   1: derived: pytest (==6.1.2)
   1: derived: ipython (==7.19.0)
   1: derived: semantic-version (>=2.8.5,<3.0.0)
   1: derived: pydantic (>=1.8.1,<2.0.0)
   1: derived: importlib-metadata (>=2.1.1,<2.2.0)
   1: derived: pywebview[cef] (==3.4)
   1: derived: PyJWT (>=1.7.1,<1.8.0)
   1: derived: keyring (>=21.8.0,<21.9.0)
   1: derived: pypac (>=0.14.0,<0.15.0)
   1: derived: pyasn1 (>=0.4.8,<0.5.0)
   1: derived: paramiko (>=2.7.2,<2.8.0)
   1: derived: azure-identity (==1.5.0b2)
   1: derived: azure-storage-blob (==12.6.0)
PyPI: 1 packages found for pywebview 3.4
   1: fact: pytest-mock (3.3.1) depends on pytest (>=5.0)
   1: selecting pytest-mock (3.3.1)
   1: fact: pytest (6.1.2) depends on attrs (>=17.4.0)
   1: fact: pytest (6.1.2) depends on iniconfig (*)
   1: fact: pytest (6.1.2) depends on packaging (*)
   1: fact: pytest (6.1.2) depends on pluggy (>=0.12,<1.0)
   1: fact: pytest (6.1.2) depends on py (>=1.8.2)
   1: fact: pytest (6.1.2) depends on toml (*)
   1: fact: pytest (6.1.2) depends on importlib-metadata (>=0.12)
   1: fact: pytest (6.1.2) depends on atomicwrites (>=1.0)
   1: fact: pytest (6.1.2) depends on colorama (*)
   1: selecting pytest (6.1.2)
   1: derived: colorama
   1: derived: atomicwrites (>=1.0)
   1: derived: toml
   1: derived: py (>=1.8.2)
   1: derived: pluggy (>=0.12,<1.0)
   1: derived: packaging
   1: derived: iniconfig
   1: derived: attrs (>=17.4.0)
   1: fact: ipython (7.19.0) depends on jedi (>=0.10)
   1: fact: ipython (7.19.0) depends on decorator (*)
   1: fact: ipython (7.19.0) depends on pickleshare (*)
   1: fact: ipython (7.19.0) depends on traitlets (>=4.2)
   1: fact: ipython (7.19.0) depends on prompt-toolkit (>=2.0.0,<3.0.0 || >3.0.0,<3.0.1 || >3.0.1,<3.1.0)
   1: fact: ipython (7.19.0) depends on pygments (*)
   1: fact: ipython (7.19.0) depends on backcall (*)
   1: fact: ipython (7.19.0) depends on pexpect (>4.3)
   1: fact: ipython (7.19.0) depends on appnope (*)
   1: fact: ipython (7.19.0) depends on colorama (*)
   1: selecting ipython (7.19.0)
   1: derived: appnope
   1: derived: pexpect (>4.3)
   1: derived: backcall
   1: derived: pygments
   1: derived: prompt-toolkit (>=2.0.0,!=3.0.0,!=3.0.1,<3.1.0)
   1: derived: traitlets (>=4.2)
   1: derived: pickleshare
   1: derived: decorator
   1: derived: jedi (>=0.10)
PyPI: 1 packages found for semantic-version >=2.8.5,<3.0.0
   1: selecting semantic-version (2.8.5)
   1: fact: pydantic (1.8.1) depends on typing-extensions (>=3.7.4.3)
   1: selecting pydantic (1.8.1)
   1: derived: typing-extensions (>=3.7.4.3)
   1: fact: importlib-metadata (2.1.1) depends on zipp (>=0.5)
   1: selecting importlib-metadata (2.1.1)
   1: derived: zipp (>=0.5)
   1: fact: pywebview (3.4) depends on pywebview (3.4)
   1: fact: pywebview (3.4) depends on pyobjc-core (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-Cocoa (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-WebKit (*)
   1: fact: pywebview (3.4) depends on PyQt5 (*)
   1: fact: pywebview (3.4) depends on pythonnet (*)
   1: fact: pywebview (3.4) depends on cefpython3 (*)
   1: selecting pywebview[cef] (3.4)
   1: derived: cefpython3
   1: derived: pythonnet
   1: derived: PyQt5
   1: derived: pyobjc-framework-WebKit
   1: derived: pyobjc-framework-Cocoa
   1: derived: pyobjc-core
   1: derived: pywebview (==3.4)
   1: selecting pyjwt (1.7.1)
   1: fact: keyring (21.8.0) depends on importlib-metadata (>=1)
   1: fact: keyring (21.8.0) depends on SecretStorage (>=3.2)
   1: fact: keyring (21.8.0) depends on jeepney (>=0.4.2)
   1: fact: keyring (21.8.0) depends on pywin32-ctypes (<0.1.0 || >0.1.0,<0.1.1 || >0.1.1)
   1: selecting keyring (21.8.0)
   1: derived: pywin32-ctypes (!=0.1.0,!=0.1.1)
   1: derived: jeepney (>=0.4.2)
   1: derived: SecretStorage (>=3.2)
   1: fact: pypac (0.14.0) depends on requests (>=2.0.0,<3.0.0)
   1: fact: pypac (0.14.0) depends on tld (>=0.9,<1.0)
   1: fact: pypac (0.14.0) depends on dukpy (>=0.2.2,<0.3.0)
   1: fact: pypac (0.14.0) depends on pyobjc-framework-SystemConfiguration (>=3.2.1)
   1: selecting pypac (0.14.0)
   1: derived: pyobjc-framework-SystemConfiguration (>=3.2.1)
   1: derived: dukpy (>=0.2.2,<0.3.0)
   1: derived: tld (>=0.9,<1.0)
   1: derived: requests (>=2.0.0,<3.0.0)
   1: selecting pyasn1 (0.4.8)
   1: fact: paramiko (2.7.2) depends on bcrypt (>=3.1.3)
   1: fact: paramiko (2.7.2) depends on cryptography (>=2.5)
   1: fact: paramiko (2.7.2) depends on pynacl (>=1.0.1)
   1: selecting paramiko (2.7.2)
   1: derived: pynacl (>=1.0.1)
   1: derived: cryptography (>=2.5)
   1: derived: bcrypt (>=3.1.3)
   1: fact: azure-identity (1.5.0b2) depends on azure-core (>=1.0.0,<2.0.0)
   1: fact: azure-identity (1.5.0b2) depends on cryptography (>=2.1.4)
   1: fact: azure-identity (1.5.0b2) depends on msal (>=1.3.0,<1.6.0)
   1: fact: azure-identity (1.5.0b2) depends on msal-extensions (>=0.3.0,<0.4.0)
   1: fact: azure-identity (1.5.0b2) depends on six (>=1.6)
   1: selecting azure-identity (1.5.0b2)
   1: derived: six (>=1.6)
   1: derived: msal-extensions (>=0.3.0,<0.4.0)
   1: derived: msal (>=1.3.0,<1.6.0)
   1: derived: azure-core (>=1.0.0,<2.0.0)
   1: fact: azure-storage-blob (12.6.0) depends on azure-core (>=1.9.0,<2.0.0)
   1: fact: azure-storage-blob (12.6.0) depends on msrest (>=0.6.10)
   1: fact: azure-storage-blob (12.6.0) depends on cryptography (>=2.1.4)
   1: selecting azure-storage-blob (12.6.0)
   1: derived: msrest (>=0.6.10)
   1: derived: azure-core (>=1.9.0,<2.0.0)
   1: selecting toml (0.10.2)
   1: selecting py (1.10.0)
   1: fact: pluggy (0.13.1) depends on importlib-metadata (>=0.12)
   1: selecting pluggy (0.13.1)
   1: fact: packaging (20.9) depends on pyparsing (>=2.0.2)
   1: selecting packaging (20.9)
   1: derived: pyparsing (>=2.0.2)
   1: selecting iniconfig (1.1.1)
   1: selecting attrs (20.3.0)
   1: selecting backcall (0.2.0)
   1: selecting pygments (2.8.1)
   1: fact: prompt-toolkit (3.0.18) depends on wcwidth (*)
   1: selecting prompt-toolkit (3.0.18)
   1: derived: wcwidth
   1: fact: traitlets (5.0.5) depends on ipython-genutils (*)
   1: selecting traitlets (5.0.5)
   1: derived: ipython-genutils
   1: selecting pickleshare (0.7.5)
   1: selecting decorator (5.0.7)
   1: fact: jedi (0.18.0) depends on parso (>=0.8.0,<0.9.0)
   1: selecting jedi (0.18.0)
   1: derived: parso (>=0.8.0,<0.9.0)
   1: selecting typing-extensions (3.7.4.3)
   1: selecting zipp (3.4.1)
   1: fact: pywebview (3.4) depends on pyobjc-core (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-Cocoa (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-WebKit (*)
   1: fact: pywebview (3.4) depends on PyQt5 (*)
   1: fact: pywebview (3.4) depends on pythonnet (*)
   1: selecting pywebview (3.4)
   1: selecting dukpy (0.2.3)
   1: selecting tld (0.12.5)
   1: fact: requests (2.25.1) depends on chardet (>=3.0.2,<5)
   1: fact: requests (2.25.1) depends on idna (>=2.5,<3)
   1: fact: requests (2.25.1) depends on urllib3 (>=1.21.1,<1.27)
   1: fact: requests (2.25.1) depends on certifi (>=2017.4.17)
   1: selecting requests (2.25.1)
   1: derived: certifi (>=2017.4.17)
   1: derived: urllib3 (>=1.21.1,<1.27)
   1: derived: idna (>=2.5,<3)
   1: derived: chardet (>=3.0.2,<5)
   1: fact: pynacl (1.4.0) depends on six (*)
   1: fact: pynacl (1.4.0) depends on cffi (>=1.4.1)
   1: selecting pynacl (1.4.0)
   1: derived: cffi (>=1.4.1)
   1: fact: cryptography (3.4.7) depends on cffi (>=1.12)
   1: selecting cryptography (3.4.7)
   1: derived: cffi (>=1.12)
   1: fact: bcrypt (3.2.0) depends on cffi (>=1.1)
   1: fact: bcrypt (3.2.0) depends on six (>=1.4.1)
   1: selecting bcrypt (3.2.0)
   1: selecting six (1.15.0)
   0: Duplicate dependencies for portalocker
   0: Different requirements found for portalocker (>=1.0,<2.0) with markers platform_system != "Windows" and portalocker (>=1.6,<2.0) with markers platform_system == "Windows".
   1: Version solving took 20.895 seconds.
   1: Tried 1 solutions.
   0: Retrying dependency resolution with the following overrides ({Package('msal-extensions', '0.3.0'): {'portalocker': <Dependency portalocker (>=1.0,<2.0)>}}).
   1: fact: poetrytest is 0.1
   1: derived: poetrytest
   1: fact: poetrytest depends on azure-storage-blob (12.6.0)
   1: fact: poetrytest depends on azure-identity (1.5.0b2)
   1: fact: poetrytest depends on paramiko (~2.7.2)
   1: fact: poetrytest depends on pyasn1 (~0.4.8)
   1: fact: poetrytest depends on pypac (~0.14.0)
   1: fact: poetrytest depends on keyring (~21.8.0)
   1: fact: poetrytest depends on PyJWT (~1.7.1)
   1: fact: poetrytest depends on pywebview (3.4)
   1: fact: poetrytest depends on importlib-metadata (~2.1.1)
   1: fact: poetrytest depends on pydantic (^1.8.1)
   1: fact: poetrytest depends on semantic-version (^2.8.5)
   1: fact: poetrytest depends on ipython (7.19.0)
   1: fact: poetrytest depends on pytest (6.1.2)
   1: fact: poetrytest depends on pytest-mock (3.3.1)
   1: fact: poetrytest depends on ipython (7.19.0)
   1: fact: poetrytest depends on pytest (6.1.2)
   1: fact: poetrytest depends on pytest-mock (3.3.1)
   1: selecting poetrytest (0.1)
   1: derived: pytest-mock (==3.3.1)
   1: derived: pytest (==6.1.2)
   1: derived: ipython (==7.19.0)
   1: derived: semantic-version (>=2.8.5,<3.0.0)
   1: derived: pydantic (>=1.8.1,<2.0.0)
   1: derived: importlib-metadata (>=2.1.1,<2.2.0)
   1: derived: pywebview[cef] (==3.4)
   1: derived: PyJWT (>=1.7.1,<1.8.0)
   1: derived: keyring (>=21.8.0,<21.9.0)
   1: derived: pypac (>=0.14.0,<0.15.0)
   1: derived: pyasn1 (>=0.4.8,<0.5.0)
   1: derived: paramiko (>=2.7.2,<2.8.0)
   1: derived: azure-identity (==1.5.0b2)
   1: derived: azure-storage-blob (==12.6.0)
   1: fact: pytest-mock (3.3.1) depends on pytest (>=5.0)
   1: selecting pytest-mock (3.3.1)
   1: fact: pytest (6.1.2) depends on attrs (>=17.4.0)
   1: fact: pytest (6.1.2) depends on iniconfig (*)
   1: fact: pytest (6.1.2) depends on packaging (*)
   1: fact: pytest (6.1.2) depends on pluggy (>=0.12,<1.0)
   1: fact: pytest (6.1.2) depends on py (>=1.8.2)
   1: fact: pytest (6.1.2) depends on toml (*)
   1: fact: pytest (6.1.2) depends on importlib-metadata (>=0.12)
   1: fact: pytest (6.1.2) depends on atomicwrites (>=1.0)
   1: fact: pytest (6.1.2) depends on colorama (*)
   1: selecting pytest (6.1.2)
   1: derived: colorama
   1: derived: atomicwrites (>=1.0)
   1: derived: toml
   1: derived: py (>=1.8.2)
   1: derived: pluggy (>=0.12,<1.0)
   1: derived: packaging
   1: derived: iniconfig
   1: derived: attrs (>=17.4.0)
   1: fact: ipython (7.19.0) depends on jedi (>=0.10)
   1: fact: ipython (7.19.0) depends on decorator (*)
   1: fact: ipython (7.19.0) depends on pickleshare (*)
   1: fact: ipython (7.19.0) depends on traitlets (>=4.2)
   1: fact: ipython (7.19.0) depends on prompt-toolkit (>=2.0.0,<3.0.0 || >3.0.0,<3.0.1 || >3.0.1,<3.1.0)
   1: fact: ipython (7.19.0) depends on pygments (*)
   1: fact: ipython (7.19.0) depends on backcall (*)
   1: fact: ipython (7.19.0) depends on pexpect (>4.3)
   1: fact: ipython (7.19.0) depends on appnope (*)
   1: fact: ipython (7.19.0) depends on colorama (*)
   1: selecting ipython (7.19.0)
   1: derived: appnope
   1: derived: pexpect (>4.3)
   1: derived: backcall
   1: derived: pygments
   1: derived: prompt-toolkit (>=2.0.0,!=3.0.0,!=3.0.1,<3.1.0)
   1: derived: traitlets (>=4.2)
   1: derived: pickleshare
   1: derived: decorator
   1: derived: jedi (>=0.10)
   1: selecting semantic-version (2.8.5)
   1: fact: pydantic (1.8.1) depends on typing-extensions (>=3.7.4.3)
   1: selecting pydantic (1.8.1)
   1: derived: typing-extensions (>=3.7.4.3)
   1: fact: importlib-metadata (2.1.1) depends on zipp (>=0.5)
   1: selecting importlib-metadata (2.1.1)
   1: derived: zipp (>=0.5)
   1: fact: pywebview (3.4) depends on pywebview (3.4)
   1: fact: pywebview (3.4) depends on pyobjc-core (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-Cocoa (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-WebKit (*)
   1: fact: pywebview (3.4) depends on PyQt5 (*)
   1: fact: pywebview (3.4) depends on pythonnet (*)
   1: fact: pywebview (3.4) depends on cefpython3 (*)
   1: selecting pywebview[cef] (3.4)
   1: derived: cefpython3
   1: derived: pythonnet
   1: derived: PyQt5
   1: derived: pyobjc-framework-WebKit
   1: derived: pyobjc-framework-Cocoa
   1: derived: pyobjc-core
   1: derived: pywebview (==3.4)
   1: selecting pyjwt (1.7.1)
   1: fact: keyring (21.8.0) depends on importlib-metadata (>=1)
   1: fact: keyring (21.8.0) depends on SecretStorage (>=3.2)
   1: fact: keyring (21.8.0) depends on jeepney (>=0.4.2)
   1: fact: keyring (21.8.0) depends on pywin32-ctypes (<0.1.0 || >0.1.0,<0.1.1 || >0.1.1)
   1: selecting keyring (21.8.0)
   1: derived: pywin32-ctypes (!=0.1.0,!=0.1.1)
   1: derived: jeepney (>=0.4.2)
   1: derived: SecretStorage (>=3.2)
   1: fact: pypac (0.14.0) depends on requests (>=2.0.0,<3.0.0)
   1: fact: pypac (0.14.0) depends on tld (>=0.9,<1.0)
   1: fact: pypac (0.14.0) depends on dukpy (>=0.2.2,<0.3.0)
   1: fact: pypac (0.14.0) depends on pyobjc-framework-SystemConfiguration (>=3.2.1)
   1: selecting pypac (0.14.0)
   1: derived: pyobjc-framework-SystemConfiguration (>=3.2.1)
   1: derived: dukpy (>=0.2.2,<0.3.0)
   1: derived: tld (>=0.9,<1.0)
   1: derived: requests (>=2.0.0,<3.0.0)
   1: selecting pyasn1 (0.4.8)
   1: fact: paramiko (2.7.2) depends on bcrypt (>=3.1.3)
   1: fact: paramiko (2.7.2) depends on cryptography (>=2.5)
   1: fact: paramiko (2.7.2) depends on pynacl (>=1.0.1)
   1: selecting paramiko (2.7.2)
   1: derived: pynacl (>=1.0.1)
   1: derived: cryptography (>=2.5)
   1: derived: bcrypt (>=3.1.3)
   1: fact: azure-identity (1.5.0b2) depends on azure-core (>=1.0.0,<2.0.0)
   1: fact: azure-identity (1.5.0b2) depends on cryptography (>=2.1.4)
   1: fact: azure-identity (1.5.0b2) depends on msal (>=1.3.0,<1.6.0)
   1: fact: azure-identity (1.5.0b2) depends on msal-extensions (>=0.3.0,<0.4.0)
   1: fact: azure-identity (1.5.0b2) depends on six (>=1.6)
   1: selecting azure-identity (1.5.0b2)
   1: derived: six (>=1.6)
   1: derived: msal-extensions (>=0.3.0,<0.4.0)
   1: derived: msal (>=1.3.0,<1.6.0)
   1: derived: azure-core (>=1.0.0,<2.0.0)
   1: fact: azure-storage-blob (12.6.0) depends on azure-core (>=1.9.0,<2.0.0)
   1: fact: azure-storage-blob (12.6.0) depends on msrest (>=0.6.10)
   1: fact: azure-storage-blob (12.6.0) depends on cryptography (>=2.1.4)
   1: selecting azure-storage-blob (12.6.0)
   1: derived: msrest (>=0.6.10)
   1: derived: azure-core (>=1.9.0,<2.0.0)
   1: selecting toml (0.10.2)
   1: selecting py (1.10.0)
   1: fact: pluggy (0.13.1) depends on importlib-metadata (>=0.12)
   1: selecting pluggy (0.13.1)
   1: fact: packaging (20.9) depends on pyparsing (>=2.0.2)
   1: selecting packaging (20.9)
   1: derived: pyparsing (>=2.0.2)
   1: selecting iniconfig (1.1.1)
   1: selecting attrs (20.3.0)
   1: selecting backcall (0.2.0)
   1: selecting pygments (2.8.1)
   1: fact: prompt-toolkit (3.0.18) depends on wcwidth (*)
   1: selecting prompt-toolkit (3.0.18)
   1: derived: wcwidth
   1: fact: traitlets (5.0.5) depends on ipython-genutils (*)
   1: selecting traitlets (5.0.5)
   1: derived: ipython-genutils
   1: selecting pickleshare (0.7.5)
   1: selecting decorator (5.0.7)
   1: fact: jedi (0.18.0) depends on parso (>=0.8.0,<0.9.0)
   1: selecting jedi (0.18.0)
   1: derived: parso (>=0.8.0,<0.9.0)
   1: selecting typing-extensions (3.7.4.3)
   1: selecting zipp (3.4.1)
   1: fact: pywebview (3.4) depends on pyobjc-core (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-Cocoa (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-WebKit (*)
   1: fact: pywebview (3.4) depends on PyQt5 (*)
   1: fact: pywebview (3.4) depends on pythonnet (*)
   1: selecting pywebview (3.4)
   1: selecting dukpy (0.2.3)
   1: selecting tld (0.12.5)
   1: fact: requests (2.25.1) depends on chardet (>=3.0.2,<5)
   1: fact: requests (2.25.1) depends on idna (>=2.5,<3)
   1: fact: requests (2.25.1) depends on urllib3 (>=1.21.1,<1.27)
   1: fact: requests (2.25.1) depends on certifi (>=2017.4.17)
   1: selecting requests (2.25.1)
   1: derived: certifi (>=2017.4.17)
   1: derived: urllib3 (>=1.21.1,<1.27)
   1: derived: idna (>=2.5,<3)
   1: derived: chardet (>=3.0.2,<5)
   1: fact: pynacl (1.4.0) depends on six (*)
   1: fact: pynacl (1.4.0) depends on cffi (>=1.4.1)
   1: selecting pynacl (1.4.0)
   1: derived: cffi (>=1.4.1)
   1: fact: cryptography (3.4.7) depends on cffi (>=1.12)
   1: selecting cryptography (3.4.7)
   1: derived: cffi (>=1.12)
   1: fact: bcrypt (3.2.0) depends on cffi (>=1.1)
   1: fact: bcrypt (3.2.0) depends on six (>=1.4.1)
   1: selecting bcrypt (3.2.0)
   1: selecting six (1.15.0)
   1: fact: msal-extensions (0.3.0) depends on msal (>=0.4.1,<2.0.0)
   1: fact: msal-extensions (0.3.0) depends on portalocker (>=1.0,<2.0)
   1: selecting msal-extensions (0.3.0)
   1: derived: portalocker (>=1.0,<2.0)
   1: fact: msal (1.5.1) depends on requests (>=2.0.0,<3)
   1: fact: msal (1.5.1) depends on PyJWT (>=1.0.0,<2)
   1: selecting msal (1.5.1)
   1: derived: PyJWT[crypto] (>=1.0.0,<2)
PyPI: 17 packages found for pyjwt >=1.0.0,<2
   1: fact: azure-core (1.13.0) depends on requests (>=2.18.4)
   1: fact: azure-core (1.13.0) depends on six (>=1.11.0)
   1: selecting azure-core (1.13.0)
   1: fact: msrest (0.6.21) depends on requests (>=2.16,<3.0)
   1: fact: msrest (0.6.21) depends on requests-oauthlib (>=0.5.0)
   1: fact: msrest (0.6.21) depends on isodate (>=0.6.0)
   1: fact: msrest (0.6.21) depends on certifi (>=2017.4.17)
   1: selecting msrest (0.6.21)
   1: derived: isodate (>=0.6.0)
   1: derived: requests-oauthlib (>=0.5.0)
   1: selecting pyparsing (2.4.7)
   1: selecting wcwidth (0.2.5)
   1: selecting ipython-genutils (0.2.0)
   1: selecting parso (0.8.2)
   1: selecting certifi (2020.12.5)
   1: selecting urllib3 (1.26.4)
   1: selecting idna (2.10)
   1: selecting chardet (4.0.0)
   1: fact: cffi (1.14.5) depends on pycparser (*)
   1: selecting cffi (1.14.5)
   1: derived: pycparser
   1: fact: isodate (0.6.0) depends on six (*)
   1: selecting isodate (0.6.0)
   1: fact: requests-oauthlib (1.3.0) depends on oauthlib (>=3.0.0)
   1: fact: requests-oauthlib (1.3.0) depends on requests (>=2.0.0)
   1: selecting requests-oauthlib (1.3.0)
   1: derived: oauthlib (>=3.0.0)
   1: selecting pycparser (2.20)
   1: selecting oauthlib (3.1.0)
   1: fact: pyjwt (1.7.1) depends on pyjwt (1.7.1)
   1: fact: pyjwt (1.7.1) depends on cryptography (>=1.4)
   1: selecting pyjwt[crypto] (1.7.1)
   1: selecting colorama (0.4.4)
   1: selecting atomicwrites (1.4.0)
   1: selecting appnope (0.1.2)
   1: fact: pexpect (4.8.0) depends on ptyprocess (>=0.5)
   1: selecting pexpect (4.8.0)
   1: derived: ptyprocess (>=0.5)
   1: selecting ptyprocess (0.7.0)
   1: selecting cefpython3 (66.1)
   1: fact: pythonnet (2.5.2) depends on pycparser (*)
   1: selecting pythonnet (2.5.2)
   1: fact: pyqt5 (5.15.4) depends on PyQt5-sip (>=12.8,<13)
   1: fact: pyqt5 (5.15.4) depends on PyQt5-Qt5 (>=5.15)
   1: selecting pyqt5 (5.15.4)
   1: derived: PyQt5-Qt5 (>=5.15)
   1: derived: PyQt5-sip (>=12.8,<13)
   1: selecting pyqt5-qt5 (5.15.2)
   1: selecting pyqt5-sip (12.8.1)
   1: fact: pyobjc-framework-webkit (7.1) depends on pyobjc-core (>=7.1)
   1: fact: pyobjc-framework-webkit (7.1) depends on pyobjc-framework-Cocoa (>=7.1)
   1: selecting pyobjc-framework-webkit (7.1)
   1: derived: pyobjc-framework-Cocoa (>=7.1)
   1: derived: pyobjc-core (>=7.1)
   1: fact: pyobjc-framework-cocoa (7.1) depends on pyobjc-core (>=7.1)
   1: selecting pyobjc-framework-cocoa (7.1)
   1: selecting pyobjc-core (7.1)
   1: selecting pywin32-ctypes (0.2.0)
   1: selecting jeepney (0.6.0)
   1: fact: secretstorage (3.3.1) depends on cryptography (>=2.0)
   1: fact: secretstorage (3.3.1) depends on jeepney (>=0.6)
   1: selecting secretstorage (3.3.1)
   1: fact: pyobjc-framework-systemconfiguration (7.1) depends on pyobjc-core (>=7.1)
   1: fact: pyobjc-framework-systemconfiguration (7.1) depends on pyobjc-framework-Cocoa (>=7.1)
   1: selecting pyobjc-framework-systemconfiguration (7.1)
   1: fact: portalocker (1.7.1) depends on pywin32 (!=226)
   1: selecting portalocker (1.7.1)
   1: derived: pywin32 (!=226)
   1: selecting pywin32 (300)
   1: Version solving took 39.117 seconds.
   1: Tried 1 solutions.
   0: Retrying dependency resolution with the following overrides ({Package('msal-extensions', '0.3.0'): {'portalocker': <Dependency portalocker (>=1.6,<2.0)>}}).
   1: fact: poetrytest is 0.1
   1: derived: poetrytest
   1: fact: poetrytest depends on azure-storage-blob (12.6.0)
   1: fact: poetrytest depends on azure-identity (1.5.0b2)
   1: fact: poetrytest depends on paramiko (~2.7.2)
   1: fact: poetrytest depends on pyasn1 (~0.4.8)
   1: fact: poetrytest depends on pypac (~0.14.0)
   1: fact: poetrytest depends on keyring (~21.8.0)
   1: fact: poetrytest depends on PyJWT (~1.7.1)
   1: fact: poetrytest depends on pywebview (3.4)
   1: fact: poetrytest depends on importlib-metadata (~2.1.1)
   1: fact: poetrytest depends on pydantic (^1.8.1)
   1: fact: poetrytest depends on semantic-version (^2.8.5)
   1: fact: poetrytest depends on ipython (7.19.0)
   1: fact: poetrytest depends on pytest (6.1.2)
   1: fact: poetrytest depends on pytest-mock (3.3.1)
   1: fact: poetrytest depends on ipython (7.19.0)
   1: fact: poetrytest depends on pytest (6.1.2)
   1: fact: poetrytest depends on pytest-mock (3.3.1)
   1: selecting poetrytest (0.1)
   1: derived: pytest-mock (==3.3.1)
   1: derived: pytest (==6.1.2)
   1: derived: ipython (==7.19.0)
   1: derived: semantic-version (>=2.8.5,<3.0.0)
   1: derived: pydantic (>=1.8.1,<2.0.0)
   1: derived: importlib-metadata (>=2.1.1,<2.2.0)
   1: derived: pywebview[cef] (==3.4)
   1: derived: PyJWT (>=1.7.1,<1.8.0)
   1: derived: keyring (>=21.8.0,<21.9.0)
   1: derived: pypac (>=0.14.0,<0.15.0)
   1: derived: pyasn1 (>=0.4.8,<0.5.0)
   1: derived: paramiko (>=2.7.2,<2.8.0)
   1: derived: azure-identity (==1.5.0b2)
   1: derived: azure-storage-blob (==12.6.0)
   1: fact: pytest-mock (3.3.1) depends on pytest (>=5.0)
   1: selecting pytest-mock (3.3.1)
   1: fact: pytest (6.1.2) depends on attrs (>=17.4.0)
   1: fact: pytest (6.1.2) depends on iniconfig (*)
   1: fact: pytest (6.1.2) depends on packaging (*)
   1: fact: pytest (6.1.2) depends on pluggy (>=0.12,<1.0)
   1: fact: pytest (6.1.2) depends on py (>=1.8.2)
   1: fact: pytest (6.1.2) depends on toml (*)
   1: fact: pytest (6.1.2) depends on importlib-metadata (>=0.12)
   1: fact: pytest (6.1.2) depends on atomicwrites (>=1.0)
   1: fact: pytest (6.1.2) depends on colorama (*)
   1: selecting pytest (6.1.2)
   1: derived: colorama
   1: derived: atomicwrites (>=1.0)
   1: derived: toml
   1: derived: py (>=1.8.2)
   1: derived: pluggy (>=0.12,<1.0)
   1: derived: packaging
   1: derived: iniconfig
   1: derived: attrs (>=17.4.0)
   1: fact: ipython (7.19.0) depends on jedi (>=0.10)
   1: fact: ipython (7.19.0) depends on decorator (*)
   1: fact: ipython (7.19.0) depends on pickleshare (*)
   1: fact: ipython (7.19.0) depends on traitlets (>=4.2)
   1: fact: ipython (7.19.0) depends on prompt-toolkit (>=2.0.0,<3.0.0 || >3.0.0,<3.0.1 || >3.0.1,<3.1.0)
   1: fact: ipython (7.19.0) depends on pygments (*)
   1: fact: ipython (7.19.0) depends on backcall (*)
   1: fact: ipython (7.19.0) depends on pexpect (>4.3)
   1: fact: ipython (7.19.0) depends on appnope (*)
   1: fact: ipython (7.19.0) depends on colorama (*)
   1: selecting ipython (7.19.0)
   1: derived: appnope
   1: derived: pexpect (>4.3)
   1: derived: backcall
   1: derived: pygments
   1: derived: prompt-toolkit (>=2.0.0,!=3.0.0,!=3.0.1,<3.1.0)
   1: derived: traitlets (>=4.2)
   1: derived: pickleshare
   1: derived: decorator
   1: derived: jedi (>=0.10)
   1: selecting semantic-version (2.8.5)
   1: fact: pydantic (1.8.1) depends on typing-extensions (>=3.7.4.3)
   1: selecting pydantic (1.8.1)
   1: derived: typing-extensions (>=3.7.4.3)
   1: fact: importlib-metadata (2.1.1) depends on zipp (>=0.5)
   1: selecting importlib-metadata (2.1.1)
   1: derived: zipp (>=0.5)
   1: fact: pywebview (3.4) depends on pywebview (3.4)
   1: fact: pywebview (3.4) depends on pyobjc-core (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-Cocoa (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-WebKit (*)
   1: fact: pywebview (3.4) depends on PyQt5 (*)
   1: fact: pywebview (3.4) depends on pythonnet (*)
   1: fact: pywebview (3.4) depends on cefpython3 (*)
   1: selecting pywebview[cef] (3.4)
   1: derived: cefpython3
   1: derived: pythonnet
   1: derived: PyQt5
   1: derived: pyobjc-framework-WebKit
   1: derived: pyobjc-framework-Cocoa
   1: derived: pyobjc-core
   1: derived: pywebview (==3.4)
   1: selecting pyjwt (1.7.1)
   1: fact: keyring (21.8.0) depends on importlib-metadata (>=1)
   1: fact: keyring (21.8.0) depends on SecretStorage (>=3.2)
   1: fact: keyring (21.8.0) depends on jeepney (>=0.4.2)
   1: fact: keyring (21.8.0) depends on pywin32-ctypes (<0.1.0 || >0.1.0,<0.1.1 || >0.1.1)
   1: selecting keyring (21.8.0)
   1: derived: pywin32-ctypes (!=0.1.0,!=0.1.1)
   1: derived: jeepney (>=0.4.2)
   1: derived: SecretStorage (>=3.2)
   1: fact: pypac (0.14.0) depends on requests (>=2.0.0,<3.0.0)
   1: fact: pypac (0.14.0) depends on tld (>=0.9,<1.0)
   1: fact: pypac (0.14.0) depends on dukpy (>=0.2.2,<0.3.0)
   1: fact: pypac (0.14.0) depends on pyobjc-framework-SystemConfiguration (>=3.2.1)
   1: selecting pypac (0.14.0)
   1: derived: pyobjc-framework-SystemConfiguration (>=3.2.1)
   1: derived: dukpy (>=0.2.2,<0.3.0)
   1: derived: tld (>=0.9,<1.0)
   1: derived: requests (>=2.0.0,<3.0.0)
   1: selecting pyasn1 (0.4.8)
   1: fact: paramiko (2.7.2) depends on bcrypt (>=3.1.3)
   1: fact: paramiko (2.7.2) depends on cryptography (>=2.5)
   1: fact: paramiko (2.7.2) depends on pynacl (>=1.0.1)
   1: selecting paramiko (2.7.2)
   1: derived: pynacl (>=1.0.1)
   1: derived: cryptography (>=2.5)
   1: derived: bcrypt (>=3.1.3)
   1: fact: azure-identity (1.5.0b2) depends on azure-core (>=1.0.0,<2.0.0)
   1: fact: azure-identity (1.5.0b2) depends on cryptography (>=2.1.4)
   1: fact: azure-identity (1.5.0b2) depends on msal (>=1.3.0,<1.6.0)
   1: fact: azure-identity (1.5.0b2) depends on msal-extensions (>=0.3.0,<0.4.0)
   1: fact: azure-identity (1.5.0b2) depends on six (>=1.6)
   1: selecting azure-identity (1.5.0b2)
   1: derived: six (>=1.6)
   1: derived: msal-extensions (>=0.3.0,<0.4.0)
   1: derived: msal (>=1.3.0,<1.6.0)
   1: derived: azure-core (>=1.0.0,<2.0.0)
   1: fact: azure-storage-blob (12.6.0) depends on azure-core (>=1.9.0,<2.0.0)
   1: fact: azure-storage-blob (12.6.0) depends on msrest (>=0.6.10)
   1: fact: azure-storage-blob (12.6.0) depends on cryptography (>=2.1.4)
   1: selecting azure-storage-blob (12.6.0)
   1: derived: msrest (>=0.6.10)
   1: derived: azure-core (>=1.9.0,<2.0.0)
   1: selecting toml (0.10.2)
   1: selecting py (1.10.0)
   1: fact: pluggy (0.13.1) depends on importlib-metadata (>=0.12)
   1: selecting pluggy (0.13.1)
   1: fact: packaging (20.9) depends on pyparsing (>=2.0.2)
   1: selecting packaging (20.9)
   1: derived: pyparsing (>=2.0.2)
   1: selecting iniconfig (1.1.1)
   1: selecting attrs (20.3.0)
   1: selecting backcall (0.2.0)
   1: selecting pygments (2.8.1)
   1: fact: prompt-toolkit (3.0.18) depends on wcwidth (*)
   1: selecting prompt-toolkit (3.0.18)
   1: derived: wcwidth
   1: fact: traitlets (5.0.5) depends on ipython-genutils (*)
   1: selecting traitlets (5.0.5)
   1: derived: ipython-genutils
   1: selecting pickleshare (0.7.5)
   1: selecting decorator (5.0.7)
   1: fact: jedi (0.18.0) depends on parso (>=0.8.0,<0.9.0)
   1: selecting jedi (0.18.0)
   1: derived: parso (>=0.8.0,<0.9.0)
   1: selecting typing-extensions (3.7.4.3)
   1: selecting zipp (3.4.1)
   1: fact: pywebview (3.4) depends on pyobjc-core (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-Cocoa (*)
   1: fact: pywebview (3.4) depends on pyobjc-framework-WebKit (*)
   1: fact: pywebview (3.4) depends on PyQt5 (*)
   1: fact: pywebview (3.4) depends on pythonnet (*)
   1: selecting pywebview (3.4)
   1: selecting dukpy (0.2.3)
   1: selecting tld (0.12.5)
   1: fact: requests (2.25.1) depends on chardet (>=3.0.2,<5)
   1: fact: requests (2.25.1) depends on idna (>=2.5,<3)
   1: fact: requests (2.25.1) depends on urllib3 (>=1.21.1,<1.27)
   1: fact: requests (2.25.1) depends on certifi (>=2017.4.17)
   1: selecting requests (2.25.1)
   1: derived: certifi (>=2017.4.17)
   1: derived: urllib3 (>=1.21.1,<1.27)
   1: derived: idna (>=2.5,<3)
   1: derived: chardet (>=3.0.2,<5)
   1: fact: pynacl (1.4.0) depends on six (*)
   1: fact: pynacl (1.4.0) depends on cffi (>=1.4.1)
   1: selecting pynacl (1.4.0)
   1: derived: cffi (>=1.4.1)
   1: fact: cryptography (3.4.7) depends on cffi (>=1.12)
   1: selecting cryptography (3.4.7)
   1: derived: cffi (>=1.12)
   1: fact: bcrypt (3.2.0) depends on cffi (>=1.1)
   1: fact: bcrypt (3.2.0) depends on six (>=1.4.1)
   1: selecting bcrypt (3.2.0)
   1: selecting six (1.15.0)
   1: fact: msal-extensions (0.3.0) depends on msal (>=0.4.1,<2.0.0)
   1: fact: msal-extensions (0.3.0) depends on portalocker (>=1.6,<2.0)
   1: selecting msal-extensions (0.3.0)
   1: derived: portalocker (>=1.6,<2.0)
   1: fact: msal (1.5.1) depends on requests (>=2.0.0,<3)
   1: fact: msal (1.5.1) depends on PyJWT (>=1.0.0,<2)
   1: selecting msal (1.5.1)
   1: derived: PyJWT[crypto] (>=1.0.0,<2)
   1: fact: azure-core (1.13.0) depends on requests (>=2.18.4)
   1: fact: azure-core (1.13.0) depends on six (>=1.11.0)
   1: selecting azure-core (1.13.0)
   1: fact: msrest (0.6.21) depends on requests (>=2.16,<3.0)
   1: fact: msrest (0.6.21) depends on requests-oauthlib (>=0.5.0)
   1: fact: msrest (0.6.21) depends on isodate (>=0.6.0)
   1: fact: msrest (0.6.21) depends on certifi (>=2017.4.17)
   1: selecting msrest (0.6.21)
   1: derived: isodate (>=0.6.0)
   1: derived: requests-oauthlib (>=0.5.0)
   1: selecting pyparsing (2.4.7)
   1: selecting wcwidth (0.2.5)
   1: selecting ipython-genutils (0.2.0)
   1: selecting parso (0.8.2)
   1: selecting certifi (2020.12.5)
   1: selecting urllib3 (1.26.4)
   1: selecting idna (2.10)
   1: selecting chardet (4.0.0)
   1: fact: cffi (1.14.5) depends on pycparser (*)
   1: selecting cffi (1.14.5)
   1: derived: pycparser
   1: fact: isodate (0.6.0) depends on six (*)
   1: selecting isodate (0.6.0)
   1: fact: requests-oauthlib (1.3.0) depends on oauthlib (>=3.0.0)
   1: fact: requests-oauthlib (1.3.0) depends on requests (>=2.0.0)
   1: selecting requests-oauthlib (1.3.0)
   1: derived: oauthlib (>=3.0.0)
   1: selecting pycparser (2.20)
   1: selecting oauthlib (3.1.0)
   1: fact: pyjwt (1.7.1) depends on pyjwt (1.7.1)
   1: fact: pyjwt (1.7.1) depends on cryptography (>=1.4)
   1: selecting pyjwt[crypto] (1.7.1)
   1: selecting colorama (0.4.4)
   1: selecting atomicwrites (1.4.0)
   1: selecting appnope (0.1.2)
   1: fact: pexpect (4.8.0) depends on ptyprocess (>=0.5)
   1: selecting pexpect (4.8.0)
   1: derived: ptyprocess (>=0.5)
   1: selecting ptyprocess (0.7.0)
   1: selecting cefpython3 (66.1)
   1: fact: pythonnet (2.5.2) depends on pycparser (*)
   1: selecting pythonnet (2.5.2)
   1: fact: pyqt5 (5.15.4) depends on PyQt5-sip (>=12.8,<13)
   1: fact: pyqt5 (5.15.4) depends on PyQt5-Qt5 (>=5.15)
   1: selecting pyqt5 (5.15.4)
   1: derived: PyQt5-Qt5 (>=5.15)
   1: derived: PyQt5-sip (>=12.8,<13)
   1: selecting pyqt5-qt5 (5.15.2)
   1: selecting pyqt5-sip (12.8.1)
   1: fact: pyobjc-framework-webkit (7.1) depends on pyobjc-core (>=7.1)
   1: fact: pyobjc-framework-webkit (7.1) depends on pyobjc-framework-Cocoa (>=7.1)
   1: selecting pyobjc-framework-webkit (7.1)
   1: derived: pyobjc-framework-Cocoa (>=7.1)
   1: derived: pyobjc-core (>=7.1)
   1: fact: pyobjc-framework-cocoa (7.1) depends on pyobjc-core (>=7.1)
   1: selecting pyobjc-framework-cocoa (7.1)
   1: selecting pyobjc-core (7.1)
   1: selecting pywin32-ctypes (0.2.0)
   1: selecting jeepney (0.6.0)
   1: fact: secretstorage (3.3.1) depends on cryptography (>=2.0)
   1: fact: secretstorage (3.3.1) depends on jeepney (>=0.6)
   1: selecting secretstorage (3.3.1)
   1: fact: pyobjc-framework-systemconfiguration (7.1) depends on pyobjc-core (>=7.1)
   1: fact: pyobjc-framework-systemconfiguration (7.1) depends on pyobjc-framework-Cocoa (>=7.1)
   1: selecting pyobjc-framework-systemconfiguration (7.1)
   1: fact: portalocker (1.7.1) depends on pywin32 (!=226)
   1: selecting portalocker (1.7.1)
   1: derived: pywin32 (!=226)
   1: selecting pywin32 (300)
   1: Version solving took 34.939 seconds.
   1: Tried 1 solutions.
   0: Complete version solving took 95.069 seconds with 2 overrides
   0: Resolved with overrides: ({Package('msal-extensions', '0.3.0'): {'portalocker': <Dependency portalocker (>=1.0,<2.0)>}}), ({Package('msal-extensions', '0.3.0'): {'portalocker': <Dependency portalocker (>=1.6,<2.0)>}})

Writing lock file

Finding the necessary packages for the current system

Package operations: 1 install, 0 updates, 0 removals, 71 skipped

  • Removing appnope (0.1.2): Pending...
  • Removing appnope (0.1.2): Skipped for the following reason: Not currently installed
  • Removing pyobjc-framework-cocoa (7.1): Pending...
  • Removing pyobjc-framework-cocoa (7.1): Skipped for the following reason: Not currently installed
  • Removing pyqt5-qt5 (5.15.2): Pending...
  • Removing pyqt5-qt5 (5.15.2): Skipped for the following reason: Not currently installed
  • Removing pexpect (4.8.0): Pending...
  • Removing pexpect (4.8.0): Skipped for the following reason: Not currently installed
  • Removing pyobjc-framework-webkit (7.1): Pending...
  • Removing pyobjc-framework-webkit (7.1): Skipped for the following reason: Not currently installed
  • Removing pyqt5 (5.15.4): Pending...
  • Removing pyqt5 (5.15.4): Skipped for the following reason: Not currently installed
  • Removing pyqt5-sip (12.8.1): Pending...
  • Removing pyqt5-sip (12.8.1): Skipped for the following reason: Not currently installed
  • Removing pyobjc-core (7.1): Pending...
  • Removing pyobjc-core (7.1): Skipped for the following reason: Not currently installed
  • Removing ptyprocess (0.7.0): Pending...
  • Removing ptyprocess (0.7.0): Skipped for the following reason: Not currently installed
  • Removing secretstorage (3.3.1): Pending...
  • Removing secretstorage (3.3.1): Skipped for the following reason: Not currently installed
  • Removing jeepney (0.6.0): Pending...
  • Removing jeepney (0.6.0): Skipped for the following reason: Not currently installed
  • Removing pyobjc-framework-systemconfiguration (7.1): Pending...
  • Removing pyobjc-framework-systemconfiguration (7.1): Skipped for the following reason: Not currently installed
  • Installing pycparser (2.20): Pending...
  • Installing pycparser (2.20): Skipped for the following reason: Already installed
  • Installing cffi (1.14.5): Pending...
  • Installing cffi (1.14.5): Skipped for the following reason: Already installed
  • Installing certifi (2020.12.5): Pending...
  • Installing certifi (2020.12.5): Skipped for the following reason: Already installed
  • Installing idna (2.10): Pending...
  • Installing idna (2.10): Skipped for the following reason: Already installed
  • Installing chardet (4.0.0): Pending...
  • Installing chardet (4.0.0): Skipped for the following reason: Already installed
  • Installing zipp (3.4.1): Pending...
  • Installing zipp (3.4.1): Skipped for the following reason: Already installed
  • Installing urllib3 (1.26.4): Pending...
  • Installing urllib3 (1.26.4): Skipped for the following reason: Already installed
  • Installing cryptography (3.4.7): Pending...
  • Installing cryptography (3.4.7): Skipped for the following reason: Already installed
  • Installing importlib-metadata (2.1.1): Pending...
  • Installing importlib-metadata (2.1.1): Skipped for the following reason: Already installed
  • Installing oauthlib (3.1.0): Pending...
  • Installing oauthlib (3.1.0): Skipped for the following reason: Already installed
  • Installing pyjwt (1.7.1): Pending...
  • Installing pyjwt (1.7.1): Skipped for the following reason: Already installed
  • Installing pywin32 (300): Pending...
  • Installing pywin32 (300): Skipped for the following reason: Already installed
  • Installing six (1.15.0): Pending...
  • Installing six (1.15.0): Skipped for the following reason: Already installed
  • Installing pyparsing (2.4.7): Pending...
  • Installing pyparsing (2.4.7): Skipped for the following reason: Already installed
  • Installing requests (2.25.1): Pending...
  • Installing requests (2.25.1): Skipped for the following reason: Already installed
  • Installing atomicwrites (1.4.0): Pending...
  • Installing atomicwrites (1.4.0): Skipped for the following reason: Already installed
  • Installing attrs (20.3.0): Pending...
  • Installing attrs (20.3.0): Skipped for the following reason: Already installed
  • Installing iniconfig (1.1.1): Pending...
  • Installing iniconfig (1.1.1): Skipped for the following reason: Already installed
  • Installing isodate (0.6.0): Pending...
  • Installing isodate (0.6.0): Skipped for the following reason: Already installed
  • Installing packaging (20.9): Pending...
  • Installing packaging (20.9): Skipped for the following reason: Already installed
  • Installing parso (0.8.2): Pending...
  • Installing parso (0.8.2): Skipped for the following reason: Already installed
  • Installing portalocker (1.7.1): Pending...
  • Installing portalocker (1.7.1): Skipped for the following reason: Already installed
  • Installing requests-oauthlib (1.3.0): Pending...
  • Installing requests-oauthlib (1.3.0): Skipped for the following reason: Already installed
  • Installing toml (0.10.2): Pending...
  • Installing toml (0.10.2): Skipped for the following reason: Already installed
  • Installing colorama (0.4.4): Pending...
  • Installing colorama (0.4.4): Skipped for the following reason: Already installed
  • Installing ipython-genutils (0.2.0): Pending...
  • Installing ipython-genutils (0.2.0): Skipped for the following reason: Already installed
  • Installing pluggy (0.13.1): Pending...
  • Installing pluggy (0.13.1): Skipped for the following reason: Already installed
  • Installing wcwidth (0.2.5): Pending...
  • Installing wcwidth (0.2.5): Skipped for the following reason: Already installed
  • Installing msal (1.5.1): Pending...
  • Installing msal (1.5.1): Skipped for the following reason: Already installed
  • Installing py (1.10.0): Pending...
  • Installing py (1.10.0): Skipped for the following reason: Already installed
  • Installing azure-core (1.13.0): Pending...
  • Installing azure-core (1.13.0): Skipped for the following reason: Already installed
  • Installing pynacl (1.4.0): Pending...
  • Installing pynacl (1.4.0): Skipped for the following reason: Already installed
  • Installing decorator (5.0.7): Pending...
  • Installing decorator (5.0.7): Skipped for the following reason: Already installed
  • Installing dukpy (0.2.3): Pending...
  • Installing dukpy (0.2.3): Skipped for the following reason: Already installed
  • Installing jedi (0.18.0): Pending...
  • Installing jedi (0.18.0): Skipped for the following reason: Already installed
  • Installing msrest (0.6.21): Pending...
  • Installing msrest (0.6.21): Skipped for the following reason: Already installed
  • Installing pickleshare (0.7.5): Pending...
  • Installing pickleshare (0.7.5): Skipped for the following reason: Already installed
  • Installing prompt-toolkit (3.0.18): Pending...
  • Installing prompt-toolkit (3.0.18): Skipped for the following reason: Already installed
  • Installing bcrypt (3.2.0): Pending...
  • Installing bcrypt (3.2.0): Skipped for the following reason: Already installed
  • Installing pythonnet (2.5.2): Pending...
  • Installing pythonnet (2.5.2): Skipped for the following reason: Already installed
  • Installing tld (0.12.5): Pending...
  • Installing tld (0.12.5): Skipped for the following reason: Already installed
  • Installing typing-extensions (3.7.4.3): Pending...
  • Installing typing-extensions (3.7.4.3): Skipped for the following reason: Already installed
  • Installing msal-extensions (0.3.0): Pending...
  • Installing msal-extensions (0.3.0): Skipped for the following reason: Already installed
  • Installing backcall (0.2.0): Pending...
  • Installing backcall (0.2.0): Skipped for the following reason: Already installed
  • Installing pytest (6.1.2): Pending...
  • Installing pytest (6.1.2): Skipped for the following reason: Already installed
  • Installing traitlets (5.0.5): Pending...
  • Installing traitlets (5.0.5): Skipped for the following reason: Already installed
  • Installing cefpython3 (66.1): Pending...
  • Installing cefpython3 (66.1): Skipped for the following reason: Already installed
  • Installing pygments (2.8.1): Pending...
  • Installing pygments (2.8.1): Skipped for the following reason: Already installed
  • Installing pywin32-ctypes (0.2.0): Pending...
  • Installing pywin32-ctypes (0.2.0): Skipped for the following reason: Already installed
  • Installing azure-identity (1.5.0b2): Pending...
  • Installing azure-identity (1.5.0b2): Skipped for the following reason: Already installed
  • Installing azure-storage-blob (12.6.0): Pending...
  • Installing azure-storage-blob (12.6.0): Skipped for the following reason: Already installed
  • Installing keyring (21.8.0): Pending...
  • Installing keyring (21.8.0): Skipped for the following reason: Already installed
  • Installing paramiko (2.7.2): Pending...
  • Installing paramiko (2.7.2): Skipped for the following reason: Already installed
  • Installing pydantic (1.8.1): Pending...
  • Installing pydantic (1.8.1): Skipped for the following reason: Already installed
  • Installing pypac (0.14.0): Pending...
  • Installing pypac (0.14.0): Skipped for the following reason: Already installed
  • Installing pytest-mock (3.3.1): Pending...
  • Installing pytest-mock (3.3.1): Skipped for the following reason: Already installed
  • Installing pywebview (3.4): Pending...
  • Installing pywebview (3.4): Skipped for the following reason: Already installed
  • Installing ipython (7.19.0): Pending...
  • Installing ipython (7.19.0): Skipped for the following reason: Already installed
  • Installing pyasn1 (0.4.8): Pending...
  • Installing pyasn1 (0.4.8): Skipped for the following reason: Already installed
  • Installing semantic-version (2.8.5): Downloading... 100%
  • Installing semantic-version (2.8.5): Installing...
  • Installing semantic-version (2.8.5)
```

**Comment by dimbleby** [comment]
yeah, that's how it works.  Most of the time poetry's cache will mean that this is fast anyway, so it doesn't matter.

I could just about conceive that someone might try optimising this specific case: but after 18 months of inactivity it's probably time to acknowlege that it's not going to happen, and close this out.

**Comment by agronholm** [comment]
You're missing the point. If my application is tested to work with a specific set of dependencies, and adding one little thing causes 47 dependencies to be upgraded to versions that have NOT been tested, that is a **problem**. Pip-tools does not have this problem, but it is otherwise inferior to Poetry.

**Comment by dimbleby** [comment]
for that workflow you can add the new dependency to `pyproject.toml` and then `poetry lock --no-update`

**Comment by dimbleby** [comment]
(fwiw I don't think that's what OP's point was at all)

**Comment by agronholm** [comment]
I was fairly sure I tried that already, but it seems this does do what I what. Thank you! This was **not** obvious from the documentation.

**Comment by neersighted** [comment]
The proposed optimization here is likely premature as the cache does make this fast except in pathological cases that likely deserve adjustments to requirements instead; however, an ask for a `poetry add --no-update` seems reasonable (or; we can make add run a minimal solve every time) -- either way, that should be a new issue.

**Comment by radoering** [comment]
> however, an ask for a `poetry add --no-update` seems reasonable (or; we can make add run a minimal solve every time) -- either way, that should be a new issue.

No need for a new option. This is how it is already done: https://github.com/python-poetry/poetry/blob/9df21a72138663e9d5a9cf80beb2f62751eed361/src/poetry/console/commands/add.py#L250

**Comment by neersighted** [comment]
Looks like this is a total non-issue then :laughing:
